kobo-uncaged/util.go

```diff
@@ -67,3 +69,29 @@ func readJSON(fn string, out interface{}) (emptyOrNotExist bool, err error) {
 
 	return false, json.NewDecoder(f).Decode(out)
 }
+
+// resizeKeepAspectRatioByExpanding resizes a sz to fill bounds while keeping
+// the aspect ratio. It is based on Qt::KeepAspectRatioByExpanding.
+func resizeKeepAspectRatioByExpanding(sz image.Point, bounds image.Point) image.Point {
+	if sz.X == 0 || sz.Y == 0 {
+		return sz
+	}
+	if rw := bounds.Y * sz.X / sz.Y; rw >= bounds.X {
+		return image.Pt(rw, bounds.Y)
+	}
+	return image.Pt(bounds.X, bounds.Y*sz.Y/sz.X)
```

Comment:
Should be `bounds.X*sz.Y/sz.X` ;)

Reply:
Fixed, but my tests still pass. I think my tests are missing some edge cases. I'm probably going to write something to generate some new ones automatically so I don't miss cases.